src/Support/Field.php

```diff
@@ -123,6 +145,11 @@ protected function getResolver(): ?Closure
                 }
             }
 
+            $fieldsAndArguments = (new ResolveInfoFieldsAndArguments($arguments[3]))->getFieldsAndArgumentsSelection($this->depth);
```

Comment:
No the big prize question: how can we avoid resolving this if there aren't going to be any validations (they're optional after all)? Do you see any way to achieve this?

Seems like a catch-22: only when we run `RulesInFields` for which we already have to extract "fields and arguments" we can then know that we wouldn't have to run it 😄

Reply:
I could probably write a function that checks if any arguments exists that would be less costly than the ResolveInfoFieldsAndArguments, and then only run ResolveInfoFieldsAndArguments if using selectfields and arguments... but not sure how much it will actually give.

Reply:
You're right, let's scrap that for now!